### ISSUE TITLE
Get-CS* commands are not working

### ISSUE THREAD

**Issue by ThoughtContagion** [MicrosoftDocs/office-docs-powershell#10777]
As I cannot seem to find the repository for this module to report the issue there, this is where it lands.

Running version 5.1.0 of the MicrosoftTeams module, every command that uses the `Get-CS*` format returns an error about the remote name not being able to be resolved.
`Get-Cs* : The remote name could not be resolved: '*.online.lync.com'`

`Get-CsOnlineUser : The remote name could not be resolved: 'webdired6.online.lync.com'`
`Get-CsApplicationAccessPolicy : The remote name could not be resolved: 'webdired6.online.lync.com'`
`Get-CsPolicyPackage : The remote name could not be resolved: 'webdired6.online.lync.com'`

Tracert, ping, Resolve-DNSName, etc. all are unable to resolve these also.

Have tried on multiple machines, different tenants, different module versions (5.0.0 and 5.1.0), on and off corporate networks, and using third-party online tools, all with the same results of not being able to find/resolve the hostname.
![image](https://user-images.githubusercontent.com/88730003/234912263-3f1a9f4f-5da8-4089-9420-71651b653426.png)

With none of these commands working, the only available commands are team specific and do not return any of the tenant level information. This limits the usefulness of this module.
`get-command "Get-*" -Module MicrosoftTeams | Where {$_.Name -notlike "Get-CS*"}`
![image](https://user-images.githubusercontent.com/88730003/234913533-f058d9f2-d729-4510-b774-87fc872248ea.png)

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 7008ec10-a056-d8bb-bb4e-b4690a84adfd
* Version Independent ID: 2dcb78d9-7f36-208e-0989-184905d73011
* Content: [Get-CsOnlineUser (SkypeForBusiness)](https://learn.microsoft.com/en-us/powershell/module/skype/get-csonlineuser?view=skype-ps)
* Content Source: [skype/skype-ps/skype/Get-CsOnlineUser.md](https://github.com/MicrosoftDocs/office-docs-powershell/blob/main/skype/skype-ps/skype/Get-CsOnlineUser.md)
* Service: **skypeforbusiness-powershell**
* GitHub Login: @isrumnon
* Microsoft Alias: **isrumnon**

**Comment by pbafna03** [comment]
Hi @ThoughtContagion, 

Which authentication mechanism is used here with Connect-MicrosoftTeams?  Are you getting same error when authenticating using just Connect-MicrosoftTeams  - the interactive user login?

**Comment by ThoughtContagion** [comment]
The authentication method used here was application auth.
I am able to run the commands using interactive authentication.

I don't believe this is a permissions issue, the application has the recommended permissions
![image](https://user-images.githubusercontent.com/88730003/234927892-a2ac8c7e-f45b-4396-a5db-ca5f0e0f1175.png)

Oddly, these commands all work using both authentication methods in PowerShell 7+, but not in version 5.1 with the same version of the module installed using application authentication.

I have just tested version 5.1.1-preview in both Windows PowerShell (v 5.1) and PowerShell Core (7.0), and again, working in Core, but not in Windows PowerShell with application authentication.

![image](https://user-images.githubusercontent.com/88730003/234928665-4eb3f239-c862-4e87-a304-726149702cd9.png)
![image](https://user-images.githubusercontent.com/88730003/234928707-892b98fb-9b65-4e1c-8bed-1c510509c497.png)

Unfortunately, I (and I'm sure others) need this to work in PowerShell 5.1.

If this is a new requirement, it should be reflected in the docs also.
![image](https://user-images.githubusercontent.com/88730003/234924884-43226afc-88b5-4980-93d2-c709b7016442.png)

**Comment by pbafna03** [comment]
@ThoughtContagion - for the app that is giving this error, can you please confirm on these 2 things:

(a) Check if by any chance you have the "application_access_custom_sba_appliance" permission assigned to Skype and Teams Tenant Admin API. If yes, remove it and post removing the permission, after a few minutes please do Disconnect-MicrosoftTeams and then Connect-MicrosoftTeams.

(b) Check if the App has a relevant [AAD Admin role](https://learn.microsoft.com/en-us/microsoftteams/using-admin-roles#teams-roles-and-capabilities) assigned to it.

**Comment by pbafna03** [comment]
> Hi @pbafna03 The application does hold the relevant roles for the uses we defined:
> 
> * Teams Administrator
> * Global Administrator
> * Exchange Administrator
> * and SharePoint Administrator
> 
> The application permissions are also restricted to only `application_access` and the defaulted `user_impersonation` ![image](https://user-images.githubusercontent.com/88730003/234935117-b4a3d96e-e931-4fa3-8465-2729965d81ee.png)
> 
> Again, this works as expected in PowerShell Core - version 7.3.4, but not in version 5.1.*
> 
> I would expect the queries to fail on all versions if the application were missing a role or permission, no?

That's right, the queries will fail irrespective of PS version if something is off with the permissions as far as the same App is used.

**Comment by pbafna03** [comment]
Hi @ThoughtContagion, 

We're not able to reproduce this, can you please raise a support ticket with the corresponding fiddler traces for the team to dig deeper here. 

Thanks,

**Comment by yogkumgit** [comment]
@ThoughtContagion Hope this comment is helpful for you. If you see a documentation update is required, please feel free to open an issue for the same. We proceed here to close it. Thanks for taking out some time to open the issue. Appreciate and encourage you to do the same in future also.

**Comment by pbafna03** [comment]
Hi @ThoughtContagion, 

Can you please remove the "application_access" permission for Skype and Teams Tenant Admin API and try this again?

Thanks,

**Comment by ThoughtContagion** [comment]
It would be beneficial if these changes were noted in the documentation somewhere. The list of required Skype/Teams permissions has shrunk in the last year or so, and each time a permission is removed, it is not documented as having been removed or required to be removed. This breaks existing applications and leaves admins scrambling to figure out what has changed. 
The Microsoft Graph module and API is fantastic for managing individual teams and applications, but does not yet provide a way to manage Tenant settings for Teams. This may make my use case and edge case, but it has consistently broken the application in use on our tenant each time this has happened.
